h/static/scripts/directive/annotation.js

```diff
@@ -72,50 +89,98 @@ function extractDocumentMetadata(model) {
   return document_;
 }
 
-/** Copy properties from viewModel into domainModel.
-*
-* All top-level properties in viewModel will be copied into domainModel,
-* overwriting any existing properties with the same keys in domainModel.
-*
-* Additionally, the `tags` property of viewModel - an array of objects
-* each with a `text` string - will become a simple array of strings in
-* domainModel.
-*
-* @param {object} domainModel The object to copy properties to
-* @param {object} viewModel The object to copy properties from
-* @returns undefined
-*
-*/
-function updateDomainModel(domainModel, viewModel) {
-  angular.extend(domainModel, viewModel);
-  domainModel.tags = viewModel.tags.map(function(tag) {
-    return tag.text;
-  });
+/** Return `true` if the given annotation is a reply, `false` otherwise. */
+function isReply(domainModel) {
+  if ((domainModel.references || []).length > 0) {
+    return true;
+  } else {
+    return false;
+  }
 }
 
-/** Update the view model from the domain model changes. */
-function updateViewModel(drafts, model, vm) {
-  var draft = drafts.get(model);
-
-  // Extend the view model with a copy of the domain model.
-  // Note that copy is used so that deep properties aren't shared.
-  vm.annotation = angular.extend({}, angular.copy(model));
+/** Return `true` if the given annotation is new, `false` otherwise.
+ *
+ * "New" means this annotation has been newly created client-side and not
+ * saved to the server yet.
+ */
+function isNew(domainModel) {
+  if (domainModel.id) {
```

Comment:
Hmm, this one I'm not so sure about it, that would work but I don't think `!!foo` is very clear